### PR TITLE
Rewrite IPlaceholderOperation to IConditionalAccessInstanceOperation …

### PR DESCRIPTION
**Pull request by 333fred**
…when in that context. Fixes https://github.com/dotnet/roslyn/issues/21671.
vso: https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20Productivity/_workitems/edit/521709

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >If syntax.IsKind(SyntaxKind.ConditionalAccessExpression) Then [](start = 12, length = 61)

The syntax check is fine for now, but we should come up with a more robust way of detecting conditional access receivers. Either keep the original issue open, or open a new one.

Comment (AlekseyTs) [reply src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: For example, we could try creating node for the placeholder when we handle BoundConditionalAccess in CreateBoundConditionalAccessOperation. The operation will be cached and this method will never be called for this bound node.

---
In reply to: [149583274](https://github.com/dotnet/roslyn/pull/23068#discussion_r149583274) [](ancestors = 149583274)

**Review by AlekseyTs — APPROVED** [review]
LGTM as a temporary solution to meet the deadline. However, see if you have time to try the suggested alternative approach.

**Comment by AlekseyTs** [comment]
Just looked at implementation of VisualBasicOperationFactory.Create, it looks like it bypasses the cache for placeholder nodes as a workaround for other issues. So, the suggested alternative approach won't work right now.

**Comment by 333fred** [comment]
@cston could I get a second review from compilers?

**Review by mavasani — APPROVED** [review]

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb
Comment (cston) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: Typo: `syntax`

File: src/Compilers/VisualBasic/Test/Semantic/Diagnostics/OperationAnalyzerTests.vb
Comment (cston) [inline src/Compilers/VisualBasic/Test/Semantic/Diagnostics/OperationAnalyzerTests.vb]: Was this test failing previously?

**Review by cston — APPROVED** [review]

File: src/Compilers/VisualBasic/Test/Semantic/Diagnostics/OperationAnalyzerTests.vb
Comment (333fred) [reply src/Compilers/VisualBasic/Test/Semantic/Diagnostics/OperationAnalyzerTests.vb]: It was not failing, but it was missing all the instance diagnostics.

**Comment by jinujoseph** [comment]
@natidea @tmat  this is good to merge once shiproom approves

**Comment by jinujoseph** [comment]
approved to merge via [Link](https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20Productivity/_workitems/edit/521709)

**Pull request merged**
